src/hooks/useReportIDs.tsx

```diff
@@ -0,0 +1,135 @@
+import React, {createContext, useCallback, useContext, useMemo} from 'react';
+import {withOnyx} from 'react-native-onyx';
+import type {OnyxCollection, OnyxEntry} from 'react-native-onyx';
+import {getCurrentUserAccountID} from '@libs/actions/Report';
+import {getPolicyMembersByIdWithoutCurrentUser} from '@libs/PolicyUtils';
+import SidebarUtils from '@libs/SidebarUtils';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Beta, Policy, PolicyMembers, PriorityMode, Report, ReportActions, TransactionViolation} from '@src/types/onyx';
+import useActiveWorkspace from './useActiveWorkspace';
+import useCurrentReportID from './useCurrentReportID';
+
+type OnyxProps = {
+    chatReports: OnyxCollection<Report>;
+    betas: OnyxEntry<Beta[]>;
+    policies: OnyxCollection<Policy>;
+    allReportActions: OnyxCollection<ReportActions>;
+    transactionViolations: OnyxCollection<TransactionViolation[]>;
+    policyMembers: OnyxCollection<PolicyMembers>;
+    priorityMode: OnyxEntry<PriorityMode>;
+};
+
+type WithReportIDsContextProviderProps = OnyxProps & {
+    children: React.ReactNode;
+    currentReportIDForTests?: string;
+};
+
+type ReportIDsContextValue = {
+    orderedReportIDs: string[];
+};
+
+const ReportIDsContext = createContext<ReportIDsContextValue>({
+    orderedReportIDs: [],
+});
+
+function WithReportIDsContextProvider({
```

Comment:
how about `OrderedReporIDsContextProvider`?

Reply:
We no longer use Higher Order Components to consume Contexts. There is no need to prefix Content Providers with `with`.

Reply:
I agree. However, regarding the naming I just have one thing to discuss. Last week, a PR got merged to main which introduced `reportIDsWithErrors` in `SidebarlinksData`. This value is then passed to `SidebarUtils.getOrderedReportIDs` to generate the ordered report IDs.

So I moved `reportIDsWithErrors` to `useReportIDs` so we can put everything that is used together in one place. That's when I renamed the hook from `useOrderedReportIDs` to `useReportIDs` and made the context export 2 values `orderedReportIDs` and `reportIDsWithErrors`.

Now if we rename this function to `OrderedReporIDsContextProvider` as you suggested, it's alright but it doesn't match with the context name `useReportIDs`.

Can we rename the function to `ReportIDsContextProviderWrapper` or something around it?

Reply:
okay your explanation makes sense and I don't have better name in mind than `WithReportIDsContextProvider`.
Let's keep the original name (i.e. `WithReportIDsContextProvider`).